### PR TITLE
WIP generic filter collector

### PR DESCRIPTION
**Pull request by fulmicoton**

### CODE REVIEW

File: src/collector/filter_collector_wrapper.rs [hunk 1]
```diff
@@ -20,99 +22,139 @@ use crate::{Score, SegmentReader, TantivyError};
...
+        let field_entry = schema.get_field_entry(self.field);
+        if !field_entry.is_fast() {
+            return Err(TantivyError::SchemaError(format!(
+                "Field {:?} is not a fast field.",
+                field_entry.name()
+            )));
+        }
+        let requested_type = TPredicateValue::to_type();
+        let field_schema_type = field_entry.field_type().value_type();
+        if requested_type != field_schema_type {
```
Comment (barrotsteindev) [inline src/collector/filter_collector_wrapper.rs:116]: Here is the run time check @fulmicoton

**Comment by fulmicoton** [comment]
@barrotsteindev LGTM. Can you add the change in CHANGELOG.md and remember to put your username.

**Comment by fulmicoton** [comment]
@barrotsteindev also can you fix the unit test?

**Comment by barrotsteindev** [comment]
Haha yeah I'm on it

**Comment by barrotsteindev** [comment]
Done @fulmicoton :)

**Comment by barrotsteindev** [comment]
Linking to issue #892

**Comment by barrotsteindev** [comment]
@fulmicoton , after looking further into `u64_lenient`, does it mean we currently only support numeric based (e.g. Date, i64 u64) fast fields?

**Comment by fulmicoton** [comment]
@barrotsteindev f64 are supported as well. 
They are probably not well bitpacked, but the principle is always the same.

Under the hood fast field are bitpacked u64 fields.
A fast monotonic mapping make the conversion between u64 <-> the actual type.

**Pull request merged**